### ISSUE TITLE
Generated config schema by Ballerina does not contain the exact type of the object

### ISSUE THREAD

**Issue by HasithaAthukorala** [ballerina-platform/ballerina-lang#35689]
**Description:**
Generated config schema[1] by Ballerina does not contain the exact type of the object.
[1] - 
```json
 "config": {
   "type": "object",
   "properties": {
      "clientId": {
        "type": "string"
      }
    }
} 
```
**Describe your solution(s)**
Include the exact type of the object in the json. As an example, type of `config` variable in the above mentioned example is `drive:ListenerConfig`

**Related Issues (optional):**
https://github.com/wso2-enterprise/choreo/issues/11859

**Comment by Dilhasha** [comment]
With the above fix, you'll get a **name** field for the record type object in the schema as follows.
e.g. https://lib.ballerina.io/ballerina/http/2.2.1/records/BearerTokenConfig

code:
`configurable http:BearerTokenConfig bearerToken = ?;`

Portion of generated config schema JSON:
```
.......
            "bearerToken": {
              "type": "object",
              "properties": {
                "token": {
                  "type": "string"
                }
              },
              "additionalProperties": false,
              "required": [
                "token"
              ],
              "name": "ballerina/http:2.2.1:BearerTokenConfig",
              "description": ""
            }
.......
```

**Comment by HasithaAthukorala** [comment]
Seems like this change is not there in `2201.0.4` . @Dilhasha could you please check and confirm?

**Comment by Dilhasha** [comment]
The fix is not available in `2201.0.4`. It will be available in `2201.1.0`. My apologies for the inconvenience caused.
